cst438_gradebook/src/main/java/com/cst438/domain/CourseRepository.java

```diff
@@ -1,7 +1,13 @@
 package com.cst438.domain;
 
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
 
 public interface CourseRepository extends CrudRepository <Course, Integer> {
+	
+	// a query to grab the id of a course if it matches the parameter passed in
+	@Query("select c from Course c where c.course_id=:course_id")
+	Course findCourseId(@Param("course_id") int course_id);
```

Comment:
May want to consider refactoring to "findCourse**By**Id" to avoid semantics confusion and maintain consistency with other repository methods. The current name may be wrongly interpreted as searching for the ID value of a course.